src/pm/hydra/ui/mpich/utils.c

```diff
@@ -1635,6 +1635,22 @@ static HYD_status parse_args(char **t_argv)
     goto fn_exit;
 }
 
+static HYD_status check_envs(void)
+{
+    HYD_status status = HYD_SUCCESS;
+    int val;
+    char *arg = "MPIEXEC_PREFIX_DEFAULT";
+
+    if (MPL_env2int("MPIEXEC_PREFIX_DEFAULT", &val)) {
+        if (val > 0) {
+            //prepend_rank_fn("prepend-rank", NULL);
+            status = HYDU_set_str(arg, &HYD_ui_info.prepend_pattern, "[%r] ");
```

Comment:
Do `HYDU_ERR_POP(status, ...)`, and add `fn_fail`, `fn_exit` labels. This way in the future when we add addition environment variable, one wouldn't easily overwritten the status return.

Reply:
When later during command-line parsing, and if additional `-l` is given and we set the string twice, does it result in memory leak? Please double check.